clearwater-infrastructure/usr/share/clearwater/bin/view-subscriber

```diff
@@ -0,0 +1,203 @@
+    except requests.exceptions.ConnectionError:
+        bindings_err = 'Unable to connect to the Sprout HTTP stack.\n' \
+                       '\tPlease contact your system administrator.'
+
+    try:
+        subscriptions_r = requests.get(subscriptions_url)
+
+        if subscriptions_r.status_code == 200:
+            try:
+                # Extract the URIs and expiry times for each subscription.
+                subscriptions_json = subscriptions_r.json()
+                subscriptions = [(subscription['req_uri'], subscription['expires']) for (subscription_id, subscription) in subscriptions_json['subscriptions'].iteritems()]
+            except ValueError:
+                subscriptions_err = 'Subscriptions information is not valid JSON.\n' \
+                                    '\tPlease contact your system administrator.'
+            except KeyError as e:
+                subscriptions_err = 'Subscriptions information returned in unexpected format ' \
+                                     '- {} key not found in JSON.\n\tPlease contact your system administrator.'.format(e)
+        elif subscriptions_r.status_code == 404:
+            # This is not an error - the subscriber may just not have any
+            # subscriptions.
+            subscriptions_err = '(none)'
+        elif subscriptions_r.status_code == 500:
+            subscriptions_err = 'Unable to contact subscriptions cache.\n' \
+                                '\tPlease contact your system administrator.'
+        else:
+            subscriptions_err = 'Unable to find subscriptions for subscriber due to unexpected ' \
+                                '{} error.\n\tPlease contact your system administrator.'.format(subscriptions_r.status_code)
+    except requests.exceptions.ConnectionError:
+        subscriptions_err = 'Unable to connect to the Sprout HTTP stack.\n' \
+                            '\tPlease contact your system administrator.'
+
+    print
+    print 'Cached data for {}'.format(impu)
+    print
+    print 'Registered bindings:'
+    if bindings_err:
+        print '\t{}'.format(bindings_err)
+    elif not bindings:
+        # We successfully parsed the JSON but there were no bindings.
+        print '\t(none)'
+    else:
+        for binding, expires in bindings:
+            print '\t- {} (expires {})'.format(binding,
+                                               datetime.datetime.fromtimestamp(expires).strftime('%c UTC'))
+
+    print
+    print 'Reg-event subscriptions:'
+    if subscriptions_err:
+        print '\t{}'.format(subscriptions_err)
+    elif not subscriptions:
+        # We successfully parsed the JSON but there were no subscriptions.
+        print '\t(none)'
+    else:
+        for subscription, expires in subscriptions:
+            print '\t- {} (expires {})'.format(subscription,
+                                               datetime.datetime.fromtimestamp(expires).strftime('%c UTC'))
+
+    print
+    print 'Subscription XML:'
```

Comment:
On second thoughts, maybe "User profile XML"? TS 29.228 describes this as the user profile and I'm worried that "Subscription" is ambiguous given that we have the reg-event subscriptions as well.